formatter.go

```diff
@@ -472,6 +497,68 @@ func (f *DefaultFormatter) formatFloatValue(value float64, bits int) string {
 	}
 }
 
+func (f *DefaultFormatter) addDigitGrouping(numStr string) string {
+	var sign string
+	if numStr[0] == '-' || numStr[0] == '+' {
+		sign = string(numStr[0])
+		numStr = numStr[1:]
+	}
+
+	parts := strings.SplitN(numStr, ".", 2)
+	intPart := f.groupDigitsInString(parts[0])
+
+	var fracPart string
+	var expPart string
+	if len(parts) == 2 {
+		parts = strings.SplitN(parts[1], "e", 2)
+		// using Reverse(parts[0]) because digit grouping in
+		// fractional part is required from Most Significant Bit
+		fracPart = "." + Reverse(f.groupDigitsInString(Reverse(parts[0])))
```

Comment:
Instead of using Reverse func, I suggest to pass direction flag (-1 or +1) to groupDigitsInString. I think the code will be easier to follow on the caller side